### PR TITLE
[RFR] ES6 and adapters

### PR DESCRIPTION
**Pull request by djhi**
- [x] rewritten using es6 with babel
- [x] added eslint airbnb configuration
- [x] introduced adapters (redis and memory) from `ratelimiter`
- [x] updated README

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "extends": "airbnb/base",
+    "env": {
+        "mocha": true,
+    },
+    "rules": {
+        "indent": [2, 4],
+        "no-multiple-empty-lines": [2, {'max': 2}],
+    },
+    "globals": {
```
Comment (Kmaschta) [inline .eslintrc:10]: Are you sure that you need these globals ?

**Comment by djhi** [comment]
I didn't make the pr on the original repo yet. 
Here are the test results: 
tests

File: .eslintrc (hunk 1 above)
Comment (djhi) [inline .eslintrc:10]: No we don't but this a default to not be annoyed when writing mocha tests

Comment (djhi) [inline .eslintrc:10]: this file should probably be in .npmignore though

**Pull request merged**